File: src/main/java/kr/co/lunatalk/domain/cartitem/dto/response/CreateCartItemResponse.java
package kr.co.lunatalk.domain.cartitem.dto.response;

import io.swagger.v3.oas.annotations.media.Schema;
import kr.co.lunatalk.domain.cartitem.domain.CartItem;

public record CreateCartItemResponse(
	@Schema(description = "등록된 cartItemId")
	Long cartItemId
) {

	public static CreateCartItemResponse from(CartItem cartItem) {
		return new CreateCartItemResponse(cartItem.getId());
	}
}

File: lunatalk-domain/src/main/kotlin/kr/co/lunatalk/domain/cartitem/dto/response/CreateCartItemResponse.kt
package kr.co.lunatalk.domain.cartitem.dto.response

import io.swagger.v3.oas.annotations.media.Schema
import kr.co.lunatalk.domain.cartitem.domain.CartItem

data class CreateCartItemResponse(
    @field:Schema(description = "등록된 cartItemId")
    val cartItemId: Long?
) {
    companion object {
        fun from(cartItem: CartItem): CreateCartItemResponse {
            return CreateCartItemResponse(cartItem.id)
        }
    }
}
